moment-shim.js

```diff
@@ -1,5 +1,44 @@
 (function() {
-/* globals define, moment */
+  /* globals define, moment */
+
+  var ComparableMoment = Ember.Object.extend(Ember.Comparable, moment.fn, {
+    compare: function(a, b) {
+      if (moment.isMoment(a) && moment.isMoment(b) && a.isBefore(b)) {
+        return -1;
+      } else if (moment.isMoment(a) && moment.isMoment(b) && a.isAfter(b)) {
+        return 1;
+      } else if (moment.isMoment(a) && !moment.isMoment(b)) {
+        return 1;
+      } else if (moment.isMoment(b)) {
+        return -1;
+      }
+
+      return 0;
+    }
+  });
+
+  var comparableMoment = function() {
+    return ComparableMoment.create(moment.apply(this, arguments));
+  };
+
+  for (var momentProp in moment) {
+    if (moment.hasOwnProperty(momentProp)) {
+      comparableMoment[momentProp] = moment[momentProp];
+    }
+  }
+
+  // Wrap global moment methods that return a full moment object
+  ['utc', 'unix'].forEach(function(method) {
```

Comment:
Sorry to nitpick, can you use Ember.EnumerableUtils.forEach here, this way IE8 will not break :)